packages/tools/devtools/devtools-core/src/data-visualization/VisualSharedTreeTypes.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { type Primitive } from "./VisualTree.js";
+
+/**
+ * Base visualizer for SharedTree.
+ */
+interface SharedTreeNodeBase {
+	schema: SharedTreeSchemaNode;
+}
+/**
+ * TODO
+ */
+interface SharedTreeSchemaNode {
+	/**
+	 * Name of the SharedTree schema.
+	 */
+	name?: string;
+
+	/**
+	 * Types allowed (e.g., string, number, boolean, handle & etc.) inside the node.
+	 */
+	allowedTypes: string;
```

Comment:
Long term, I think this probably wants to be a tree itself. Something like:
```suggestion
	allowedTypes: SharedTreeSchemaNode[];
```

That way in the view, you can see each type option, and drill into it to get more details.

Reply:
But just a string is probably fine for v1